File: ios-oc-first/ios/OCFirst/ViewController.m
//
//  ViewController.m
//  OCFirst
//
//  Created by Damien Cavailles on 16/02/2017.
//  Copyright © 2017 Damien Cavailles. All rights reserved.
//

#import "ViewController.h"
#import <React/RCTRootView.h>
#import "KerKerNative.h"

@interface ViewController ()

@end

NSString *const KerKerOutputReceive = @"KerKerOutput";

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self renderReact];
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(outputFromReact:)
                                                 name:KerKerOutputReceive
                                               object:nil];
    
}

-(void)outputFromReact:(NSNotification*)notification {
    NSString* message = notification.userInfo[@"message"];
    [self.outputField setText:[NSString stringWithFormat:@"%@\n%@", self.outputField.text, message]];
    
}

- (IBAction)onToucheLeBoutton:(id)sender {
    [KerKerNative increment];
}

-(void)renderReact {
    NSLog(@"renderReact");
    NSURL *jsCodeLocation = [NSURL
                             URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios"];
    RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL : jsCodeLocation
                         moduleName        : @"KerKer"
                         initialProperties :
     @{
       @"scores" : @[
               @{
                   @"name" : @"Alex",
                   @"value": @"42"
                   },
               @{
                   @"name" : @"Joel",
                   @"value": @"10"
                   }
               ]
       }
                          launchOptions    : nil];
    
    rootView.frame = CGRectMake(0, 0, 320, 240);
    [self.moustache addSubview:rootView];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: ios-first/ios/iOSFirst/ViewController.swift
//
//  ViewController.swift
//  iOSFirst
//
//  Created by Damien Cavailles on 16/02/2017.
//  Copyright © 2017 Damien Cavailles. All rights reserved.
//

import UIKit
import React


class ViewController: UIViewController {

    @IBOutlet weak var outputTextView: UITextView!
    @IBOutlet weak var moustache: UIView!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        renderReact()
        NotificationCenter.default.addObserver(self, selector: #selector(self.reactOutput(_:)), name: NSNotification.Name(rawValue: "KerKerNativeOutput"), object: nil);
    }
    
    @IBAction func onTouchUpInside(_ sender: Any) {
        KerKerNative.increment()
    }

    func reactOutput(_ notification : NSNotification) {
        let message = notification.userInfo?["message"] as! String;
        self.outputTextView.text = self.outputTextView.text + "\n" +  message
    }
    
    
    
    func renderReact() {
        
        let jsCodeLocation = URL(string: "http://localhost:8081/index.ios.bundle?platform=ios")
        let mockData:NSDictionary = ["scores":
            [
                ["name":"Alex", "value":"42"],
                ["name":"Joel", "value":"10"]
            ]
        ]
        
        let rootView : RCTRootView = RCTRootView(
            bundleURL: jsCodeLocation,
            moduleName: "KerKer",
            initialProperties: mockData as [NSObject : AnyObject],
            launchOptions: nil
        )
        rootView.frame = CGRect(x: 0,
                                y: 0,
                                width: self.moustache.frame.size.width,
                                height: self.moustache.frame.size.height);
        
        self.moustache.addSubview(rootView)
        
    }

}

